dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java

```diff
@@ -207,17 +216,59 @@ public Response getAllMessages (
             @Context final HttpServletResponse response,
             @PathParam("language") final String language){
 
-        final long languageId = "default".equalsIgnoreCase(language) ? APILocator.getLanguageAPI().getDefaultLanguage().getId() : LanguageUtil.getLanguageId(language);
-        final Language language1 = APILocator.getLanguageAPI().getLanguage(languageId);
-        if(!UtilMethods.isSet(language1)){
-            final String message = "Language with Id or Code: "+ language + " does not exist";
-            Logger.error(this,message);
-            throw new IllegalArgumentException(message);
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredAnonAccess(AnonymousAccess.READ)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(false).init();
+
+        final User user = initData.getUser();
+
+        final Locale locale = "default".equalsIgnoreCase(language) ? APILocator.getLanguageAPI().getDefaultLanguage().asLocale() : ConversionUtils.toLocale(language);
+        final Locale[] locales = LanguageUtil.getAvailableLocales();
+        for(int i=0;i<locales.length;i++){
+            final Locale currentLocale = locales[i];
+            if(currentLocale.getLanguage().equalsIgnoreCase(locale.getLanguage())){
+                if(UtilMethods.isSet(locale.getCountry())){
+                    if(currentLocale.getCountry().equalsIgnoreCase(locale.getCountry())){
+                        break;
+                    }
+                }else{
+                    break;
+                }
+
+            } else if(i == (locales.length-1)){
+                final String message = "Locale: "+ locale + " not found in Portal.properties file";
+                Logger.error(this,message);
+                throw new DoesNotExistException(message);
+            }
+        }
+
+        //Messages in the properties file
+        final Map mapPropertiesFile = LanguageUtil.getAllMessagesByLocale(locale);
+
+        final Map result = new TreeMap(mapPropertiesFile);
+
+        final Language language1 = APILocator.getLanguageAPI().getLanguage(locale.getLanguage(),locale.getCountry());
+        if(UtilMethods.isSet(language1)) {
+            //Language Keys
+            final Map mapLanguageKeys = APILocator.getLanguageAPI()
+                    .getLanguageKeys(locale.getLanguage()).stream().collect(
+                            Collectors.toMap(LanguageKey::getKey, LanguageKey::getValue));
+
+            result.putAll(mapLanguageKeys);
+
+            //Language Variable
+            long langId = language1.getId();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'langId' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5671019)